tools/mixxx-isolate.py

```diff
@@ -0,0 +1,496 @@
+        if not args.isolate:
+            helperArgs.append("--no-isolate-ht")
+
+    if args.debug:
+        helperArgs.append("--debug")
+
+    command = selectCommand(args)
+    if not command:
+        return None
+
+    fullCommand = command + helperArgs
+    log.info(" Execute %s", fullCommand)
+
+    return subprocess.Popen(
+        fullCommand,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=sys.stderr,
+    )
+
+
+def startHelper(args):
+    """Execute $self with adjusted arguments under privilege elevation"""
+
+    helper = elevatePermissions(args)
+    mixxxProc = None
+    if not helper:
+        log.critical(" Could not spawn with elevated permissions")
+    while True:
+        try:
+            helper.poll()
+            if helper.returncode == 127:
+                # user aborted password request, nothing happened so it's
+                # save to exit
+                log.error(" Could not run elevate process. Aborting")
+                sys.exit(1)
+            elif helper.returncode == 23:
+                # Error setting up isolation
+                log.critical(" Error creating isolation. Aborting")
+                sys.exit(1)
+            elif helper.returncode is not None:
+                # helper process died
+                log.info(helper.stdout.read(-1))
+                log.error(
+                    " Helper process died: %s."
+                    " Cleanup might be required required",
+                    helper.returncode,
+                )
+                break
+
+            line = helper.stdout.readline().decode("utf-8", "ignore")
+            if line.startswith("STATUS:"):
+                status = line[7:].strip()
+                print("GOT STATUS:" + status)
+                if status == "ISOLATION":
+                    log.info(" Isolation setup complete. Starting mixxx...")
+                    mixxxProc = subprocess.Popen(
+                        [args.mixxx, "--engine-cpu-set", args.cpuset_engine]
+                        + args.mixxx_args
+                    )
+                    mixxxProc.wait()
+                    log.info(" Mixxx finished")
+                    helper.stdin.write("exit\n".encode("ASCII"))
+                    try:
+                        # this might likely not work
+                        helper.send_signal(signal.SIGUSR1)
+                    except PermissionError:
+                        pass
+                    break
+                elif status == "CLEAN":
+                    # cleanup finished.
+                    log.info(" Exit successfully")
+                    sys.exit(0)
+            else:
+                print(line)
+
+        except KeyboardInterrupt:
+            log.info(" Received keyboard interrupt. Stopping")
+            if mixxxProc and mixxxProc.returncode is None:
+                mixxxProc.send_signal(signal.SIGINT)
+            break
+        except Exception as e:
+            log.exception(e)
```

Comment:
```suggestion
        except:
            log.exception("An error occurred")
```

Also, there is a risk of infinite loop